x/compliance/handler_test.go

```diff
@@ -662,75 +692,48 @@ func TestHandler_CertifyRevokedModel(t *testing.T) {
 		_, err := setup.Handler(setup.Ctx, revokeModelMsg)
 		require.NoError(t, err)
 
-		receivedComplianceInfo, _ := queryComplianceInfo(
-			setup,
-			testconstants.Vid,
-			testconstants.Pid,
-			testconstants.SoftwareVersion,
-			certificationType,
-		)
-		checkRevokedModelInfo(t, revokeModelMsg, receivedComplianceInfo)
-
 		// cancel model revocation
 		certifyModelMsg := NewMsgCertifyModel(
-			testconstants.Vid,
-			testconstants.Pid,
-			testconstants.SoftwareVersion,
-			testconstants.SoftwareVersionString,
-			certificationType,
-			setup.CertificationCenter,
-		)
-		certifyModelMsg.CertificationDate = revocationTime.AddDate(0, 0, 1).Format(time.RFC3339)
+			vid, pid, softwareVersion, softwareVersionString, certificationType, setup.CertificationCenter)
+		certifyModelMsg.CertificationDate = time.Now().UTC().Format(time.RFC3339)
 		_, err = setup.Handler(setup.Ctx, certifyModelMsg)
 		require.NoError(t, err)
-	}
-}
-
-func TestHandler_CheckCertificationDone(t *testing.T) {
-	setup := Setup(t)
 
-	// add model version
-	vid, pid, softwareVersion, softwareVersionString := setup.AddModelVersion(
-		testconstants.Vid, testconstants.Pid, testconstants.SoftwareVersion, testconstants.SoftwareVersionString)
+		// query certified model info
+		receivedComplianceInfo, _ := queryComplianceInfo(setup, vid, pid, softwareVersion, certificationType)
 
-	// add testing results
-	setup.AddTestingResults(vid, pid, softwareVersion, softwareVersionString)
+		// check
+		checkCertifiedModelInfo(t, certifyModelMsg, receivedComplianceInfo)
+		require.Equal(t, 1, len(receivedComplianceInfo.History))
 
-	for _, certificationType := range setup.CertificationTypes {
-		// certify model
-		certifyModelMsg := NewMsgCertifyModel(
-			vid, pid, softwareVersion, softwareVersionString, certificationType, setup.CertificationCenter)
-		_, err := setup.Handler(setup.Ctx, certifyModelMsg)
-		require.NoError(t, err)
+		require.Equal(t, types.CodeRevoked, receivedComplianceInfo.History[0].SoftwareVersionCertificationStatus)
+		require.Equal(t, revokeModelMsg.RevocationDate, receivedComplianceInfo.History[0].Date)
 
-		// create another certification center account
-		accAddress := GenerateAccAddress()
-		setup.AddAccount(accAddress, []dclauthtypes.AccountRole{dclauthtypes.CertificationCenter})
+		// query certified model
+		certifiedModel, _ := queryCertifiedModel(setup, vid, pid, softwareVersion, certificationType)
+		require.True(t, certifiedModel.Value)
 
-		// try to certify model again from new account
-		secondCertifyModelMsg := NewMsgCertifyModel(
-			vid, pid, softwareVersion, softwareVersionString, certificationType, accAddress)
-		_, err = setup.Handler(setup.Ctx, secondCertifyModelMsg)
-		require.Error(t, err)
-		require.True(t, types.ErrAlreadyCertified.Is(err))
+		// query revoked model
+		revokedModel, _ := queryRevokedModel(setup, vid, pid, softwareVersion, certificationType)
+		require.False(t, revokedModel.Value)
 	}
 }
 
```

Comment:
Do we have a test that checks that a non-existent model can not be revoked?

Reply:
The test has been added in scope of https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/182